code/modules/spells/spell.dm

```diff
@@ -246,7 +246,7 @@
 		return target // They're just standing around, proceed as normal
 
 	if(HAS_TRAIT(cast_loc, TRAIT_CASTABLE_LOC))
-		return cast_loc // They're in an atom which allows casting, so redirect the caster to loc
+		return ismob(cast_loc.loc) ? cast_loc.loc : cast_loc // They're in an atom which allows casting, so redirect the caster to loc
```

Comment:
I'm a little bit wary of of far we're extending the scope of what is considered the "caster" here, like this is basically saying "if a PAI casts Ethereal Jaunt while in someone's pocket then it actually casts Ethereal Jaunt on the guy holding the PAI" which starts to no longer make sense.

Reply:
I'll add another trait then, TRAIT_SPELLS_TRANSFER_TO_LOC